### DISCUSSION TITLE
Hadronic cascade amplitudes in json

### DISCUSSION THREAD

**Discussion by mmikhasenko** [hep-statistics-standard/hep-statistics-standard#34] [Technical]
In this thread, I would like to discuss how hadronic decay models can be serialized as json files in a way similar to HS3.
Perhaps, in the future, we get this format to merge to HS3 and reach out large community of amplitude-analysis-practitioners.

In hadron spectroscopy,
we study probability distribution describing exclusive low-multiplicity reactions (1->3, 1->4 decays, 2->2, 2->3, 2->4 productions). The probability distribution is modelled as unnormalized function in n-dimentions,

```
unpolarized_intensity(tau) = sum_{spin_proj} | A (tau, spin_proj) |^2
```
where `tau` are kinematic variables,
`spin_proj` are either integers or half integers (`two_spin_proj` would be integers).

Let’s take an example on how to describe A for a decay 1->3 is modelled by a cascade reaction,
The particles are numbered 0,1,2,3 for the mother particle, and decay particles respectively.

### Decay chains

0->(12)+3 with resonance A12
- `production_coupling_scheme`: `LS` with L=0, S=1/2,
- `coefficient` is 0.3+0.1i

0->(12)+3 with resonance A12
- `production_coupling_scheme`: `LS` with L=1, S=1/2,
- `coefficient` is 0.3+0.1i

0->(12)+3 with resonance B12
- `production_coupling_scheme`: `LS` with L=2, S=1/2,
- `coefficient` is 0.5-1.1i

0->1+(23) with resonance A23
- `production_coupling_scheme`: `LS` with L=1, S=3/2,
- `coefficient` is 1.5

### General information:

- `masses`: m0=,…
- `spins`: j0=1/2, j1=1/2, j2=0, j3=0
- `quantization reference chain`: 0((12)3)

### Description of the resonances and their decays:

- A12 with 
  - `spin` 1/2
  - `lineshape`: BreitWigner(m_A12,G_A12).
  - `decay_coupling_schene`: `byParity` with lambda_a=1/2, lambda_b=0, `ηηηphaseisplus`=false 

- B12 with 
  - spin 3/2
  - `lineshape`: BreitWigner(m_B12,G_B12). 
  - `decay_coupling_schene`: `byParity` with lambda_a=1/2, lambda_b=0, `ηηηphaseisplus`=false

- A23 with
  - spin 2
  - `lineshape` BreitWigner(m_A23,G_A23). The production coefficient is 1.5
  - `decay_coupling_schene`: `byParity` with lambda_a=0, lambda_b=0, `ηηηphaseisplus`=true

**Comment by mmikhasenko** [comment]
@cburgard let's follow up on the discussion we had after your talk at our [recent meeting](https://indico.cern.ch/event/1348003/contributions/5677083/)

What would be the HS3-like representation of it?
I am wondering how the overall object should look like, i.e. shou
```
 - name
 - type: "UnpolarizedDecayDensity"
 - amplitude: "MyDecayAmplitude"
```

followed by
```
"MyDecayAmplitude":
 - type: "1->3 decay"
 - system_mass: 1.1 or distribution
 - final_state_particle_masses: [0.1, 0.2, 0.3]
 - system_spins: ...
 - quantization reference chain:
 - decay chains: [`mychain12_A`, `mychain12_B`, ...]
```
followed by a description of decay chains,
followed by a description of subchannel resonances and their decays.

Let's write a json file for the example I placed above and discuss if it is general enough.

**Comment by jonas-eschle** [comment]
Hi @mmikhasenko , thanks for opening this thread! I've chatted with a few people in the past about this and how to best incorporate it as I think it's a very useful addition, indeed. To me, it seems probably something similar like the "histfactory" part: a sublanguage that can be decomposed.

I think it should be a separate language that can be sticked into it (and HS3 is designed to exactly allow for these things. However, agreeing with amplitude analysis people on a common language (and definition!) surely is it's own endeavor, but one that should be tackled. I think Remco had some ideas/efforts in this direction.

At the same time, the full model should be able to be natively put into HS3 with a sympy/mathematica/etc-like format as a custom distribution.

**Reply by mmikhasenko** [reply]
Adding Remco to the thread, just talked to him

**Comment by cburgard** [comment]
Hi @mmikhasenko !
Thanks a lot for raising this interesting point.
I fully agree with both of you that we should make sure to have a way to accurately and precisely describe such cases in HS3.
Unfortunately, my domain knowledge for usage of elaborate decay chains in statistical models is severely limited, so I would need to rely on one of you to come up with an initial draft, but I'm happy to give opinions and suggestions.

More on point: The first two decay chains you list in your example seem identical to me, what's the point here? Do we need to support such cases?

**Comment by mmikhasenko** [comment]
> More on point: The first two decay chains you list in your example seem identical to me, what's the point here? Do we need to support such cases?

They are the same by a typo, thanks for spotting. Modifying to have different L

**Comment by mmikhasenko** [comment]
@cburgard @jonas-eschle  thanks for the reply.
Let's try to wrap my example into a valid HS3 for the start, I would try to parse it and see what's missing

**Comment by mmikhasenko** [comment]
Just talked to @redeboer, we've got zero version of the format proposal

**Comment by mmikhasenko** [comment]
FYI
https://github.com/mmikhasenko/AmplitudeModelSerialization?tab=readme-ov-file

**Comment by cburgard** [comment]
Hi Misha!
Thanks a lot for this extensive proposal!
I've only gone over it quickly for now and will try to take more time to look at it in detail later, but maybe some preliminary comments already now:

- I see that you already have some julia code in your repo, but I'm having a hard time figuring out what it does exactly (not a julia person here). Are these mapping to some specific Julia implementation already, or is this just code that generates the json file without any functional backend?
- I'd be very interested adopting this format for HS3, but exactly how the configuration maps to a Pdf is not clear to me. I see that you have this brief math section in the beginning where you outline that it's a product of Breit-Wigners, but how all the details given later affect the parameters of these is a little bit opaque to me still still (but again, I'm not an expert in hadron physics, so maybe I'm the one to blame here). Maybe you could supplement the explanation a little bit until someone (like me) would be able to also write importers and exporters for e.g. ROOT for this format? Otherwise, we could also try and schedule a meeting to discuss, but eventually, we will need a complete write-up anyway.

**Reply by mmikhasenko** [reply]
Thanks for the comment.

I just added an initial section on how amplitude is related to pdf. 


https://github.com/mmikhasenko/AmplitudeModelSerialization/blob/main/description.md#amplitude-model-and-observables


It seems to me that all of this objects will sit in the distributions/functions section of the json

**Comment by mmikhasenko** [comment]
The `ThreeBodyDecays.jl` is now able to write the serialized models,

I just added an example to 
https://github.com/mmikhasenko/AmplitudeModelSerialization

I would be great to discuss how we move to common notations, keys
